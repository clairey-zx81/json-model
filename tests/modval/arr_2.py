# Generated for model: ./arr_2.model.json
#
# Generated by jmc (JSON Model Compiler) (https://github.com/clairey-zx81/json-model)
#

from typing import Callable
import re as re
import datetime
import urllib.parse

type Jsonable = None|bool|int|float|str|list[Jsonable]|dict[str, Jsonable]
type CheckFun = Callable[[Jsonable, str], bool]
type PropMap = dict[str, CheckFun]
type TagMap = dict[None|bool|float|int|str, CheckFun]

def is_valid_re(value: Jsonable, path: str) -> bool:
    if isinstance(value, str):
        try:
            re.compile(value)
            return True
        except:
            return False
    return False

def is_valid_date(value: Jsonable, path: str) -> bool:
    if isinstance(value, str):
        try:
            datetime.date.fromisoformat(value)
            return True
        except:
            return False
    return False

def is_valid_url(value: Jsonable, path: str) -> bool:
    if isinstance(value, str):
        try:
            urllib.parse.urlparse(value)
            return True
        except:
            return False
    return False


# define "$" ($)
def json_model_0(value: Jsonable, path: str) -> bool:
    # $
    result = isinstance(value, list) and len(value) == 2
    if result:
        # $.0
        result = isinstance(value[0], bool)
        if result:
            # $.1
            result = isinstance(value[1], int) and not isinstance(value[1], bool) and value[1] >= 0
    return result

# entry function check_model
def check_model(value: Jsonable, path: str = "$") -> bool:
    return json_model_0(value, path)


