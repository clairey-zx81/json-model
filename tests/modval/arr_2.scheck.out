arr_2.0.true.json: PASS
arr_2.1.false.json: FAIL ['not array or bad array length at $ [$]', 'not array or unexpected array at $ [$]']
arr_2.2.false.json: FAIL ['not array or bad array length at $ [$]', 'not array or unexpected array at $ [$]']
arr_2.3.false.json: FAIL ['not array or bad array length at $ [$]', 'not array or unexpected array at $ [$]']
arr_2.4.false.json: FAIL ['not array or bad array length at $ [$]', 'not array or unexpected array at $ [$]']
arr_2.5.false.json: FAIL ['not array or bad array length at $ [$]', 'not array or unexpected array at $ [$]']
